### PR TITLE
Clean up code 59

### PR DESCRIPTION
**Pull request by ktbolt**
These are some code clean cleaning changes I made for https://github.com/StanfordCBCL/svZeroDPlus/issues/59.

Most of these are simple changes.

I also added improved error checking in the applications `main` function.

### CODE REVIEW

**Comment by mrp089** [comment]
@ktbolt, I think something in merge 7182a1f1d8537dbcf6044bd54e4ec05dcf0c0789 went wrong

**Comment by ktbolt** [comment]
@mrp089 Yes, it looks odd. Shall I close this and merge from the command line?

**Comment by mrp089** [comment]
@ktbolt, yeah, that's probably easier. I think the build error is just some variable naming.

**Comment by mrp089** [comment]
You can still keep this pull request open. Whatever you commit to this branch will automatically show up here.

File: applications/svzerodsolver.cpp [hunk 1]
```diff
@@ -56,38 +56,56 @@ int main(int argc, char* argv[]) {
...
+  nlohmann::json config;
+
+  try { 
+    config = nlohmann::json::parse(input_file);
+
+  } catch (const nlohmann::json::parse_error& e) {
+    std::cout << "[svzerodsolver] Error: Parsing the input file '" << input_file_name << "' has failed." << std::endl;
+    std::cout << "[svzerodsolver] Details of the parsing error: " << std::endl;
+    std::cout << e.what() << std::endl;
+    return 1;
```
Comment (mrp089) [inline applications/svzerodsolver.cpp:103]: I appreciate all the error handling! Why do we never throw an actual error?

File: src/algebra/SparseSystem.h [hunk 2]
```diff
@@ -96,7 +96,7 @@ class SparseSystem {
    *
    * @param n Size of the system
    */
-  SparseSystem(unsigned int n);
+  SparseSystem(int n);
```
Comment (mrp089) [inline src/algebra/SparseSystem.h:99]: Why did you remove a couple of `unsigned`?

File: src/interface/interface.cpp [hunk 3]
```diff
@@ -61,34 +61,32 @@ extern "C" void initialize(std::string input_file, int& problem_id,
                            std::vector<std::string>& block_names,
                            std::vector<std::string>& variable_names);
 
-extern "C" void set_external_step_size(const int problem_id,
+extern "C" void set_external_step_size(int problem_id,
```
Comment (mrp089) [inline src/interface/interface.cpp:64]: Why did you remove a couple of `const`?

**Review by mrp089 — APPROVED** [review]
@ktbolt looks good! I just had three questions for my understanding.

File: src/algebra/SparseSystem.h (hunk 2 above)
Comment (ktbolt) [reply src/algebra/SparseSystem.h:99]: Unsigned int's are really only used when doing something with bit values. If you set an unsigned int to a negative value then you get garbage. 

You can argue that it can be self-documenting, that `n` should be >= 0. But isn't it better to actually use a better name such as `system_size` that documents what it is rather than `n`?

File: src/interface/interface.cpp (hunk 3 above)
Comment (ktbolt) [reply src/interface/interface.cpp:64]: An int is passed by value so you don't need to `const` qualify it, doesn't make a difference to the caller.
